portfolio/src/main/java/com/google/sps/data/Comments.java

```diff
@@ -32,6 +32,40 @@ public Comment(Date date, String text, long rating) {
             this.text = text;
             this.rating = rating;
         }
+
+        public Date getDate() {
+            return date;
+        }
+
+        public String getText() {
+            return text;
+        }
+
+        /**
+         * Gets the comment from comment entity.
+         * @param entity    Entity of the "Comment" kind
+         * @return A Comment object with fields retrieved from the entity.
+         */
+        public static Comment getCommentFromEntity(Entity entity) {
+            Date date = (Date) entity.getProperty(DATE);
+            String text = (String) entity.getProperty(TEXT);
+            long rating = (long) entity.getProperty(RATING);
+            return new Comment(date, text, rating);
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (obj == null || getClass() != obj.getClass()) {
+                return false;
+            }
+            Comment otherComment = (Comment) obj;
+            return this.date.equals(otherComment.date)
+                    && this.rating == otherComment.rating
+                    && this.text.equals(otherComment.text);
```

Comment:
If date or text can be null, that could be a problem. You can use Objects.equals to test equality while handling null at the same time